frontends/systolic-lang/gen_post_op.py

```diff
@@ -0,0 +1,329 @@
+                row_ready_wire.in_ = (valid_signal | row_ready_reg.out) @ 1
+                # Logic to hold the systolic array output values. We need registers
+                # because the output values for systolic arrays are only available
+                # for one cycle before they change.
+                val_ready.in_ = (
+                    valid_signal
+                    & (
+                        idx_signal
+                        == cb.ExprBuilder(py_ast.ConstantPort(addr_width, col))
+                    )
+                ) @ 1
+                wire_value.in_ = val_ready.out @ value_signal
+                wire_value.in_ = ~(val_ready.out) @ reg_value.out
+                reg_value.in_ = val_ready.out @ value_signal
+                reg_value.write_en = val_ready.out @ 1
+
+    # Helper function adds assignment wire.in = reg.out == col ? pe_{row}_{col}_out.
+    def build_assignment(group: cb.GroupBuilder, wire, register, output_val):
+        group.asgn(
+            wire.port("in"),
+            output_val.out,
+            register.port("out") == cb.ExprBuilder(py_ast.ConstantPort(BITWIDTH, col)),
+        )
+
+    group = comp.static_group(f"r{row}_helper", 1)
+
+    # Current value we are performing relu on.
+    cur_val = comp.wire(f"r{row}_cur_val", BITWIDTH)
+    # Current idx within the row (i.e., column) for the value we are performing relu on.
+    idx_reg = comp.reg(f"r{row}_cur_idx", BITWIDTH)
+    # Handling logic to hold the systolic array's output values so they're available
+    # for moer than one cycle.
+    store_output_vals(comp, row, num_cols, addr_width)
+    for col in range(num_cols):
+        output_val = comp.get_cell(f"r{row}_c{col}_val_wire")
+        # Assigning to cur_val wire so that we always have the current value of the
+        # row based on idx_reg.
+        build_assignment(group, cur_val, idx_reg, output_val)
+
+    # Instantiate an instance of a leaky_relu component
+    relu_instance = comp.cell(f"leaky_relu_r{row}", py_ast.CompInst("leaky_relu", []))
+    # Wire that tells us we are finished with relu operation for this row.
+    relu_finished_wire = comp.wire(f"relu_finished_wire_r{row}", 1)
+
+    row_ready_wire = comp.get_cell(f"r{row}_ready_wire")
+
+    # Annoying memory stuff because we can't use ref cells
+    this = comp.this()
+    mem_name = OUT_MEM + f"_{row}"
+    addr0_port = cb.ExprBuilder.unwrap(this.port(mem_name + "_addr0"))
+    relu_addr0_port = relu_instance.port(OUT_MEM + "_addr0")
+    mem_write_data_port = cb.ExprBuilder.unwrap(this.port(mem_name + "_write_data"))
+    relu_write_data_port = relu_instance.port(OUT_MEM + "_write_data")
+    mem_write_en_port = cb.ExprBuilder.unwrap(this.port(mem_name + "_write_en"))
+    relu_write_en_port = relu_instance.port(OUT_MEM + "_write_en")
+    mem_done_port = this.port(mem_name + "_done")
+    relu_done_port = cb.ExprBuilder.unwrap(relu_instance.port(OUT_MEM + "_done"))
```

Comment:
Can we hoist this into a function? I imagine this is the code that "splats" the interface signals of the memory into the component. This is probably a generally useful thing to do and maybe should be added into the `builder` library.